src/main/java/Duke.java

```diff
@@ -1,10 +1,255 @@
+                markTask(userInput, taskList);
+                break;
+            case UNMARK:
+                unmarkTask(userInput, taskList);
+                break;
+            case DELETE:
+                deleteTask(userInput, taskList);
+                break;
+            default:
+                throw new IllegalArgumentException("Hey, please choose from the following commands\n" +
+                        "if you want to add task, please use todo, deadline or event\n" +
+                        "if you want to mark or unmark task, please use mark or unmark\n" +
+                        "if you want delete a task, please use delete\n" +
+                        "if you want to view the existing task list, please enter list.");
+        }
+    }
+
+    private static void markTask(String userInput, List<Task> taskList) {
+        String[] parts = userInput.split(" ");
+        if (parts.length < 2 || parts[1].trim().isEmpty()) {
+            throw new IllegalArgumentException("Please provide a task number.");
+        }
+        int taskIndex = Integer.parseInt(parts[1]) - 1;
+        if (taskIndex < 0 || taskIndex >= taskList.size()) {
+            throw new IndexOutOfBoundsException("Please check how many tasks are there in your list.");
+        }
+        taskList.get(taskIndex).markAsDone();
+        System.out.println("Nice, I've marked this task as done for you:");
+        System.out.println((taskList.get(taskIndex)));
+    }
+
+    private static void unmarkTask(String userInput, List<Task> taskList) {
+        String[] parts = userInput.split(" ");
+        if (parts.length < 2 || parts[1].trim().isEmpty()) {
+            throw new IllegalArgumentException("Please provide a task number.");
+        }
+        int taskIndex = Integer.parseInt(parts[1]) - 1;
+        if (taskIndex < 0 || taskIndex >= taskList.size()) {
+            throw new IndexOutOfBoundsException("Please check how many tasks are there in your list.");
+        }
+        taskList.get(taskIndex).markAsUndone();
+        System.out.println("Nice, I've marked this task as undone for you:");
+        System.out.println((taskList.get(taskIndex)));
+    }
+    private static void addTask(String userInput, List<Task> taskList) throws Exception {
+        String[] parts = userInput.split(" ", 2);
+        if (parts.length < 2 || parts[1].trim().isEmpty()) {
+            throw new IllegalArgumentException("The description of a todo task cannot be empty.");
+        }
+        String command = parts[0];
+        String taskInfo = parts[1];
+
+        Task newTask;
+
+        switch (command) {
+            case "todo":
+                newTask = new ToDo(taskInfo);
+                break;
+            case "deadline":
+                String[] details = taskInfo.split("/by", 2);
+                if (details.length < 2) {
+                    throw new IllegalArgumentException("Invalid format for deadline, please provide a deadline using /by.");
```

Comment:
Perhaps wrap lines so they do not exceed the [120 character limit](https://se-education.org/guides/conventions/java/intermediate.html#layout)?